src/Storages/Streaming/StorageMaterializedView.cpp

```diff
@@ -73,6 +74,49 @@ String generateInnerQueryID(const StorageID & view_id)
         return ".inner.query-id.from-" + toString(view_id.uuid);
     return ".inner.query-id.from-" + view_id.getTableName();
 }
+
+void resetSNsOfStreamingSources(const auto & streaming_sources, const std::vector<Int64> & sns)
+{
+    if (sns.size() != streaming_sources.size()) [[unlikely]]
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Number of sequence numbers doesn't match number of streaming sources");
+
+    for (auto [streaming_source, sn] : std::views::zip(streaming_sources, sns))
```

Comment:
`auto &` ?

Reply:
`const auto &` or `auto &` will get error 
```
error: loop variable '[streaming_source, sn]' binds to a temporary value produced by a range of type 'decltype(zip_view<all_t<const vector<shared_ptr<ISource>, allocator<shared_ptr<ISource>>> &>, all_t<const vector<long long, allocator<long long>> &>>(std::forward<const std::vector<std::shared_ptr<DB::Streaming::ISource>> &>(__rs), std::forward<const std::vector<long long> &>(__rs)))' (aka 'zip_view<ref_view<const std::vector<std::shared_ptr<DB::Streaming::ISource>, std::allocator<std::shared_ptr<DB::Streaming::ISource>>>>, ref_view<const std::vector<long long, std::allocator<long long>>>>') [-Werror,-Wrange-loop-bind-reference]
        for (const auto & [streaming_source, sn] : std::views::zip(streaming_sources, sns))
```
 in fact, `auto` also means a reference , you can see https://quuxplusone.github.io/blog/2020/08/26/wrange-loop-analysis/
 
 or `auto &&`